vars/neoDeploy.groovy

```diff
@@ -8,222 +11,218 @@ import com.sap.piper.tools.ToolDescriptor
-                script.currentBuild.setResult('UNSTABLE')
-                echo "[WARNING][$STEP_NAME] Build has been set to unstable since old config framework is used."
-            }
-        }
-
-        // Backward compatibility end
+        Map stepCompatibilityConfiguration = handleCompatibility(script, parameters)
 
         // load default & individual configuration
         Map configuration = ConfigurationHelper.newInstance(this)
             .loadStepDefaults()
             .mixinGeneralConfig(script.commonPipelineEnvironment, GENERAL_CONFIG_KEYS)
             .mixin(stepCompatibilityConfiguration)
-            .mixinStepConfig(script.commonPipelineEnvironment, STEP_CONFIG_KEYS)
-            .mixinStageConfig(script.commonPipelineEnvironment, parameters.stageName?:env.STAGE_NAME, STEP_CONFIG_KEYS)
-            .addIfEmpty('archivePath', script.commonPipelineEnvironment.getMtarFilePath())
-            .mixin(parameters, PARAMETER_KEYS)
+            .mixinStepConfig(script.commonPipelineEnvironment, STEP_CONFIG_KEYS, CONFIG_KEY_COMPATIBILITY)
+            .mixinStageConfig(script.commonPipelineEnvironment, parameters.stageName ?: env.STAGE_NAME, STEP_CONFIG_KEYS, CONFIG_KEY_COMPATIBILITY)
+            .addIfEmpty('source', script.commonPipelineEnvironment.getMtarFilePath())
+            .mixin(parameters, PARAMETER_KEYS, CONFIG_KEY_COMPATIBILITY)
+            .withMandatoryProperty('neo', 'Error in neoDeploy: neo not configured.')
+            .withMandatoryProperty('source', 'Error in neoDeploy: source not configured.')
+            .withMandatoryProperty('deployMode', 'Error in neoDeploy: deployMode not configured.')
+            .withMandatoryProperty('warAction', 'Error in neoDeploy: warAction not configured.')
             .use()
 
         utils.pushToSWA([
-            step: STEP_NAME,
-            stepParam1: configuration.deployMode == 'mta'?'mta':'war', // ['mta', 'warParams', 'warPropertiesFile']
-            stepParam2: configuration.warAction == 'rolling-update'?'blue-green':'standard', // ['deploy', 'deploy-mta', 'rolling-update']
+            step      : STEP_NAME,
+            stepParam1: configuration.deployMode == 'mta' ? 'mta' : 'war', // ['mta', 'warParams', 'warPropertiesFile']
+            stepParam2: configuration.warAction == 'rolling-update' ? 'blue-green' : 'standard', // ['deploy', 'deploy-mta', 'rolling-update']
             stepParam3: parameters?.script == null,
-            stepParam4: ! stepCompatibilityConfiguration.isEmpty(),
+            stepParam4: !stepCompatibilityConfiguration.isEmpty(),
         ], configuration)
 
-        def archivePath = configuration.archivePath
-        if(archivePath?.trim()) {
-            if (!fileExists(archivePath)) {
-                error "Archive cannot be found with parameter archivePath: '${archivePath}'."
+        ToolDescriptor neo = new ToolDescriptor('SAP Cloud Platform Console Client', 'NEO_HOME', 'neoHome', '/tools/', 'neo.sh', null, 'version')
```

Comment:
I would like to suggest to remove the ToolDescriptor approach. This was introduced as result of some heavy discussions about "should we validate the build server setup or not, but I think the world has turned in the meantime. We rely more and more on docker images where we can sure about the environment. Even more I would like to suggest to not provide the configuration options regarding the location of the neo toolset. Instead the build server should be setup properly having the tools in the path so that we can simply call e.g. `neo.sh`.